Track.h

```diff
@@ -56,32 +57,44 @@ class Track
-
   // Non-const versions needed for CopyOut of Matriplex.
   SVector6&     parameters_nc() {return state_.parameters;}
   SMatrixSym66& errors_nc()     {return state_.errors;}
   TrackState&   state_nc()      {return state_;}
-
-  SVector3 position() const {return SVector3(state_.parameters[0],state_.parameters[1],state_.parameters[2]);}
-  SVector3 momentum() const {return SVector3(state_.parameters[3],state_.parameters[4],state_.parameters[5]);}
-
-  int      charge() const {return state_.charge;}
-  float    chi2()   const {return chi2_;}
   int      label()  const {return label_;}
 
-  float posPhi() const { return getPhi(state_.parameters[0],state_.parameters[1]); }
-  float momPhi() const { return getPhi(state_.parameters[3],state_.parameters[4]); }
-  float posEta() const { return getEta(state_.parameters[0],state_.parameters[1],state_.parameters[2]); }
-  float momEta() const { return getEta(state_.parameters[3],state_.parameters[4],state_.parameters[5]); }
-
-  float posR()   const { return getHypot(state_.parameters[0],state_.parameters[1]); }
-  float pT()     const { return getHypot(state_.parameters[3],state_.parameters[4]); }
+  // track state position 
+  const float radius() const {return std::sqrt(getRad2(state_.parameters[0],state_.parameters[1]));}
+  const float x()      const {return state_.parameters[0];}
+  const float y()      const {return state_.parameters[1];}
+  const float z()      const {return state_.parameters[2];}
+  const float posPhi() const {return getPhi(state_.parameters[0],state_.parameters[1]);}
+  const float posEta() const {return getEta(radius(),z());}
+
+  // track state momentum
+  const float pt()     const {return std::sqrt(getRad2(state_.parameters[3],state_.parameters[4]));}
+  const float px()     const {return state_.parameters[3];}
+  const float py()     const {return state_.parameters[4];}
+  const float pz()     const {return state_.parameters[5];}
+  const float momPhi() const {return getPhi(state_.parameters[3],state_.parameters[4]);}
+  const float momEta() const {return getEta(pt(),pz());}
+
+  // track state momentum errors
+  const float ept()     const {return std::sqrt(std::abs(getRadErr2(state_.parameters[3],state_.parameters[4],state_.errors[3][3],state_.errors[4][4],state_.errors[3][4])));}
+  const float epx()     const {return std::sqrt(state_.errors[3][3]);}
+  const float epy()     const {return std::sqrt(state_.errors[4][4]);}
+  const float epz()     const {return std::sqrt(state_.errors[5][5]);}
+  const float emomPhi() const {return std::sqrt(std::abs(getPhiErr2(state_.parameters[3],state_.parameters[4],state_.errors[3][3],state_.errors[4][4],state_.errors[3][4])));}
+  const float emomEta() const {return std::sqrt(std::abs(getEtaErr2(state_.parameters[3],state_.parameters[4],state_.parameters[5],state_.errors[3][3],state_.errors[4][4],state_.errors[5][5],state_.errors[3][4],state_.errors[3][5],state_.errors[4][5])));}
```

Comment:
removed first const where appropriate (i.e. for functions I previously defined const type const {return state.value} --> only for by value... kept const type const for return by reference